### PR TITLE
diagnostics:add cpu arch

### PR DESCRIPTION
**Pull request by Max-Cheng**
Signed-off-by: Max-Cheng <aaron9shire@gmail.com>

What's Changed:

This PR implements the Diagnostics service server information retrieving interface, which is used to retrieve server load / hardware / system wide information. And this PR includes the following information:

- Hardware
  - CPU 
    - Arch

### Check List 

Tests 

- Unit test

### Release note 
- Support display CPU arch

### CODE REVIEW

File: src/server/service/diagnostics/sys.rs [hunk 1]
```diff
@@ -305,6 +305,16 @@ fn cpu_hardware_info(collector: &mut Vec<ServerInfoItem>) {
...
         ("cpu-vendor-id", processor.get_vendor_id().to_string()),
     ];
+    //Depend on Rust lib return CPU arch not matching
+    //Golang lib so need this match loop to conversion
+    //GOlang Doc:https://go.googlesource.com/go/+/go1.15.8/src/runtime/internal/sys/zgoarch_amd64.go#7
+    //Rust Doc:http://web.mit.edu/rust-lang_v1.26.0/arch/amd64_ubuntu1404/share/doc/rust/html/std/env/consts/constant.ARCH.html
+    let arch = match std::env::consts::ARCH {
+        "x86_64" => "amd64",
+        "x86" => "386",
+        _ => std::env::consts::ARCH,
```
Comment (breeswish) [inline src/server/service/diagnostics/sys.rs:315]: Do we need to unify arm64 to arm?

**Review by breeswish — COMMENTED** [review]
Thanks! Left a few comments

**Comment by breeswish** [comment]
Some unit tests failed. You may need to update these tests. You can run tests via `make dev`.

File: src/server/service/diagnostics/sys.rs [hunk 2]
```diff
@@ -305,6 +305,16 @@ fn cpu_hardware_info(collector: &mut Vec<ServerInfoItem>) {
         ("cpu-frequency", format!("{}MHz", processor.get_frequency())),
         ("cpu-vendor-id", processor.get_vendor_id().to_string()),
     ];
+    //Depend on Rust lib return CPU arch not matching
+    //Golang lib so need this match loop to conversion
+    //GOlang Doc:https://go.googlesource.com/go/+/go1.15.8/src/runtime/internal/sys/zgoarch_amd64.go#7
+    //Rust Doc:http://web.mit.edu/rust-lang_v1.26.0/arch/amd64_ubuntu1404/share/doc/rust/html/std/env/consts/constant.ARCH.html
```
Comment (sticnarf) [inline src/server/service/diagnostics/sys.rs:311]: Style suggestion: add one space after `//`

File: src/server/service/diagnostics/sys.rs [hunk 3]
```diff
@@ -305,6 +305,16 @@ fn cpu_hardware_info(collector: &mut Vec<ServerInfoItem>) {
         ("cpu-frequency", format!("{}MHz", processor.get_frequency())),
         ("cpu-vendor-id", processor.get_vendor_id().to_string()),
     ];
+    //Depend on Rust lib return CPU arch not matching
+    //Golang lib so need this match loop to conversion
+    //GOlang Doc:https://go.googlesource.com/go/+/go1.15.8/src/runtime/internal/sys/zgoarch_amd64.go#7
```
Comment (sticnarf) [inline src/server/service/diagnostics/sys.rs:310]: GOlang -> Golang

**Comment by breeswish** [comment]
Thanks for the contribution. We requires sign-off for all commits. You can use git rebase to rewrite all commits to be signed-off, or you can simply `git diff` against master, then re-create a brand new commit for it.

**Comment by Max-Cheng** [comment]
Seem like this? @breeswish

**Review by breeswish — APPROVED** [review]
Yup, looks good!

**Comment by breeswish** [comment]
@crazycs520 PTAL

**Review by sticnarf — APPROVED** [review]

**Comment by Max-Cheng** [comment]
@breeswish Tks

**Pull request merged**